starknet_py/utils/data_transformer/data_transformer.py

```diff
@@ -386,11 +396,16 @@ def from_python(
         calldata: List[int] = []
         for name, cairo_type in type_by_name.items():
             if name not in named_arguments:
-                raise TypeError(f"Input {name} not provided.")
+                raise CairoSerializerException(f"Input {name} not provided.")
 
-            values = self.resolve_type(cairo_type).from_python(
-                cairo_type, name, named_arguments[name]
-            )
+            try:
+                values = self.resolve_type(cairo_type).from_python(
+                    cairo_type, name, named_arguments[name]
+                )
+            except ValueError as err:
+                raise InvalidDataException(str(err)) from err
+            except TypeError as err:
+                raise CairoSerializerException(str(err)) from err
```

Comment:
What kind of exceptions do we expect here?

Reply:
I'm not sure if I understand you correctly 😅 We expect errors from transformer classes (TupleTransformer, FeltTransformer). I've changed errors in the data_transformer.py, but I don't want to do it in e.g. Felt.py (it throws ValueErrors)

Reply:
Those exceptions might not be very informative for users, but we can keep it this way.